packages/backend-common/src/service/lib/ServiceBuilderImpl.ts

```diff
@@ -120,34 +154,96 @@ export class ServiceBuilderImpl implements ServiceBuilder {
         reject(e);
       });
 
-      const server = stoppable(
-        app.listen(port, host, () => {
-          logger.info(`Listening on ${host}:${port}`);
-        }),
-        0,
-      );
+      let server: http.Server;
+
+      if (httpsSettings) {
+        logger.info('Initializing https server');
+
+        let credentials: { key: string; cert: string };
+        const signingOptions: any = httpsSettings?.certificate;
+
+        if (signingOptions?.algorithm !== undefined) {
+          const certificateAttributes: Array<any> = [];
+
+          Object.getOwnPropertyNames(signingOptions?.attributes).forEach(
+            propertyName => {
+              const propertyValue = (signingOptions?.attributes as any)[
+                propertyName
+              ];
+
+              certificateAttributes.push({
+                name: propertyName,
+                value: propertyValue,
+              });
+            },
+          );
+
+          // TODO: Create a type def for selfsigned.
+          const signatures = require('selfsigned').generate(
+            certificateAttributes,
+            {
+              keySize: signingOptions?.size || 2048,
+              algorithm: signingOptions?.algorithm || 'sha256',
+              days: signingOptions?.days || 30,
+            },
+          );
+
+          credentials = { key: signatures.private, cert: signatures.cert };
+        } else {
+          credentials = { key: '', cert: '' };
+
+          const keyPath = path.resolve(__dirname, signingOptions?.key);
```

Comment:
I think the expected behaviour here is that the certificate path is relative to the config file, so imo it's better to rely on the config secrets to load in certificates from files instead. Using the example in the previous PR you'd just get the entire certificate file loaded into the config `key` and `cert` values.